buildSrc/src/test/java/org/elasticsearch/gradle/VersionTests.java

```diff
@@ -97,6 +101,20 @@ public void testExceptionSyntax() {
         Version.fromString("foo.bar.baz");
     }
 
+    public void testLabels() {
+        Version v = Version.fromString("1.2.3");
+        assertThat(v.getLabels(), empty());
+
+        v = Version.fromString("1.2.3-rc1");
+        assertThat(v.getLabels(), contains("rc1"));
+
+        v = Version.fromString("1.2.3-rc1-SNAPSHOT");
```

Comment:
It's unlikely in practice, but it's supported by the pattern in `Version`, so I'm testing it.

Reply:
I'm inclined to _remove_ support for this. It also makes `consumers` more complex because they have to deal with `labels` being a collection rather than just being a simple literal.

Reply:
Version now just has a simple `qualifier` field.